run/test/specs/group_disappearing_messages_gif.spec.ts

```diff
@@ -1,98 +1,100 @@
-  // Select GIF tab
-  await clickOnCoordinates(device1, InteractionPoints.GifButtonKeyboardClosed);
-  // Need to select Continue on GIF warning
-  await device1.clickOnByAccessibilityID("Continue", 5000);
-  await device1.clickOnElementAll({
-    strategy: "xpath",
-    selector: XPATHS.FIRST_GIF,
-  });
-  await device1.clickOnByAccessibilityID("Text input box");
-  await device1.inputText("accessibility id", "Text input box", testMessage);
-  await device1.clickOnByAccessibilityID("Send button");
-  await device1.waitForTextElementToBePresent({
-    strategy: "accessibility id",
-    selector: "Message sent status: Sent",
-    maxWait: 20000,
-  });
-  await Promise.all([
-    device2.waitForTextElementToBePresent({
-      strategy: "accessibility id",
-      selector: "Message body",
-      text: testMessage,
-    }),
-    device3.waitForTextElementToBePresent({
-      strategy: "accessibility id",
-      selector: "Message body",
-      text: testMessage,
-    }),
-  ]);
+  await device1.sendGIF(testMessage);
+  if (platform === 'ios') {
+    await Promise.all([
+      device2.waitForTextElementToBePresent({
+        strategy: 'accessibility id',
+        selector: 'Message body',
+        text: testMessage,
+      }),
+      device3.waitForTextElementToBePresent({
+        strategy: 'accessibility id',
+        selector: 'Message body',
+        text: testMessage,
+      }),
+    ]);
+  }
+  if (platform === 'android') {
+    await Promise.all([
+      device2.waitForTextElementToBePresent({
+        strategy: 'accessibility id',
+        selector: 'Media message',
+      }),
+      device3.waitForTextElementToBePresent({
+        strategy: 'accessibility id',
+        selector: 'Media message',
+      }),
+    ]);
+  }
   // Wait for 10 seconds
-  await sleepFor(10000);
+  await sleepFor(30000);
   // Check if GIF has been deleted on both devices
-  await Promise.all([
-    device1.hasElementBeenDeleted({
-      strategy: "accessibility id",
-      selector: "Message body",
-      maxWait: 1000,
-      text: testMessage,
-    }),
-    device2.hasElementBeenDeleted({
-      strategy: "accessibility id",
-      selector: "Message body",
-      maxWait: 1000,
-      text: testMessage,
-    }),
-    device3.hasElementBeenDeleted({
-      strategy: "accessibility id",
-      selector: "Message body",
-      maxWait: 1000,
-      text: testMessage,
-    }),
-  ]);
+  if (platform === 'ios') {
+    await Promise.all([
+      device1.hasElementBeenDeleted({
+        strategy: 'accessibility id',
+        selector: 'Message body',
+        maxWait: 1000,
+        text: testMessage,
+      }),
+      device2.hasElementBeenDeleted({
+        strategy: 'accessibility id',
+        selector: 'Message body',
+        maxWait: 1000,
+        text: testMessage,
+      }),
+      device3.hasElementBeenDeleted({
+        strategy: 'accessibility id',
+        selector: 'Message body',
+        maxWait: 1000,
+        text: testMessage,
+      }),
+    ]);
```

Comment:
for all of those (not just the one I highlighted) you can do
```suggestion
   const deletedArgs = {
        strategy: 'accessibility id',
        selector: 'Message body',
        maxWait: 1000,
        text: testMessage,
      };
    await Promise.all([
      device1.hasElementBeenDeleted(deletedArgs),
      device2.hasElementBeenDeleted(deletedArgs),
      device3.hasElementBeenDeleted(deletedArgs),
    ]);
```

Reply:
or even

```ts
 const deletedArgs = {
        strategy: 'accessibility id',
        selector: 'Message body',
        maxWait: 1000,
        text: testMessage,
      };
    await Promise.all([device1, device2, device3].map(device =>
      device.hasElementBeenDeleted(deletedArgs))
    );
```

Reply:
updated to 
```
await Promise.all(
      [device1, device2, device3].map(device =>
        device.hasElementBeenDeleted({
          strategy: 'accessibility id',
          selector: 'Media message',
          maxWait: 1000,
        })
      )
    );
```